### ISSUE TITLE
[explicit-function-return-type] allowTypedFunctionExpressions flag should ignore class methods written as arrow functions

### ISSUE THREAD

**Issue by ayush000** [typescript-eslint/typescript-eslint#484]
**Repro**

```JSON
"rules": {
    "@typescript-eslint/explicit-function-return-type": [1, {
      "allowTypedFunctionExpressions": true
    }]
  },
```

```TS
type MethodType = () => void;

class App {
  private method: MethodType = () => {}
}
```

**Expected Result**
The `explicit-function-return-type` rule (with `allowTypedFunctionExpressions=true`) should not complain/be triggered by the code above.

**Actual Result**
I get the warning "Missing return type on function." for the private method

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.7.0` |
| `@typescript-eslint/parser`        | `1.7.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `12.0.0` |
| `yarn`                              | `1.15.2` |
